tests/core/test_datamodules.py

```diff
@@ -533,6 +534,56 @@ def test_simple_hyperparameters_saving():
     assert data.hparams == AttributeDict({"arg0": 10, "arg1": "foo", "kwarg0": "bar"})
 
 
+def test_define_as_dataclass():
+    # makes sure that no functionality is broken and the user can still manually make
+    # super().__init__ call with parameters
+    # also tests all the dataclass features that can be enabled without breaking anything
+    @dataclass(init=True, repr=True, eq=True, order=True, unsafe_hash=True, frozen=False)
+    class BoringDataModule1(LightningDataModule):
+        batch_size: int = 32
```

Comment:
I pulled the branch and tried it out with batch size as required

```suggestion
        batch_size: int
```

But it fails

Reply:
Not sure what you mean. Removing the default value for `batch_size`wouldn't cause any problem.  If you are trying a different batch size for `BoringDataModule3`, then that's a caveat with `SklearnDataModule`.

![Screenshot](https://user-images.githubusercontent.com/40473118/131467084-4a697b4a-dd6c-4453-a062-72cf2f397895.png)
And in case you are facing some problem like this, this is just how `dataclasses` create the `init` method for subclasses derived from other `dataclass`.